netket/experimental/driver/infidelity_sr.py

```diff
@@ -0,0 +1,497 @@
+from typing import Any
+from collections.abc import Callable
+
+import jax
+from jax.flatten_util import ravel_pytree
+import flax
+
+from netket import jax as nkjax
+from netket.optimizer.solver import cholesky
+from netket.vqs import MCState, FullSumState, VariationalState
+from netket.utils import timing, struct
+from netket.utils.types import ScalarOrSchedule, Optimizer, Array, PyTree
+from netket.jax._jacobian.default_mode import JacobianMode
+from netket.operator import AbstractOperator
+from netket import stats as nkstats
+from netket.jax import HashablePartial
+
+from netket.driver.abstract_variational_driver import (
+    AbstractVariationalDriver,
+)
+from netket._src.ngd.sr_srt_common import sr, srt
+from netket._src.ngd.srt_onthefly import srt_onthefly
+from netket.experimental.operator.infidelity.expect import get_local_estimator
+
+ApplyFun = Callable[[PyTree, Array], Array]
+KernelArgs = tuple[ApplyFun, PyTree, Array, tuple[Any, ...]]
+KernelFun = Callable[[PyTree, Array], KernelArgs]
+DeriativesArgs = tuple[ApplyFun, PyTree, PyTree, Array]
+
+
+@jax.jit
+def _flatten_samples(x):
+    # return x.reshape(-1, x.shape[-1])
+    return jax.lax.collapse(x, 0, x.ndim - 1)
+
+
+class Infidelity_SR(AbstractVariationalDriver):
+    r"""
+    Infidelity minimization with respect to a target state :math:`|Φ⟩` (with possibly an operator :math:`U` such as :math:`|Φ⟩ \equiv U|Φ⟩`)
+    using Variational Monte Carlo (VMC) and **Stochastic Reconfiguration/Natural Gradient Descent**.
+    This driver is mathematically equivalent to the standard :class:`netket.driver.VMC` with the
+    preconditioner :class:`netket.optimizer.SR(solver=netket.optimizer.solvers.cholesky) <netket.optimizer.SR>`,
+    but can easily switch between the standard and the kernel/minSR formulation of Natural Gradient Descent.
+    In this case, the effective Hamiltonian is the projector onto the target state :math:`|Φ⟩`, namely :math:`H = |Φ⟩⟨Φ| / ⟨Φ|Φ⟩`.
+    The variational state is indicated as :math:`|ψ⟩`.
+    - The standard formulation computes the updates as:
+
+    .. math::
```

Comment:
you can delete this, just link to the other driver. maybe put the stuff from infidelity in entket_fidelity as a docstrig?